src/NodeAutoCompleteViewExtension/ViewModels/NodeAutoCompleteBarViewModel.cs

```diff
@@ -887,7 +898,7 @@ internal void PopulateAutoComplete()
                             Title = x.Description,  
                             Probability = x.Score.ToString(),
                             EntryNodeIndex = 0,
-                            EntryNodeInPort = x.PortToConnect,
+                            EntryNodeInPort = PortViewModel.PortType == PortType.Output ? x.PortToConnect : 0, //single autocomplete service doesnt specify port for input
```

Comment:
So I guess this should be -1 to give a chance to prevent a wrong usage when port is Input ? 
Also maybe  it's worth initializing here EntryNodeOutPort to 0 explicitly for clarity.